localtypings/pxtarget.d.ts

```diff
@@ -37,6 +37,10 @@ declare namespace pxt {
         bannedRepos?: string[];
         allowUnapproved?: boolean;
         preferredRepos?: string[]; // list of company/project(#tag) of packages to show by default in search
+        // format:
+        // "acme-corp/pxt-widget": "min:v0.1.2" - auto-upgrade to that version
+        // "acme-corp/pxt-widget": "dv:foo,bar" - add "disablesVariant": ["foo", "bar"] to pxt.json
+        upgrades?: pxt.Map<string>;
```

Comment:
do we want to have a mini format like or just have a JSON structure to describe those ranages? what happens if you want to combine min:v0.1.2,dv:foo,bar.

For the target, we have something like this:         
```
patches?: pxt.Map<UpgradePolicy[]>; // semver range -> upgrade policies
```

Reply:
I'm not a great fan of using JSON as a DSL... in this case there's also no need to combine - if you enforce a minimum version, you also make sure it has the right properties in its pxt.json

Reply:
In fact I think the package min version should probably move to UpgradePolicy. Currently, we show the wrong version number in explorer after upgrade.

Reply:
OK, I'm mistaken. I'll just apply it in the place where UpgradePolicies are applied.

Reply:
OK done. Package rules are now still in targetconfig.json so they can be updated out of band. The package refs are updated in pxt.json and persisted, and disabledVariants are done on the fly.